src/ray/raylet/worker_pool.h

```diff
@@ -585,9 +608,17 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
   absl::flat_hash_map<Language, State, std::hash<int>> states_by_lang_;
 
   /// The pool of idle non-actor workers of all languages. This is used to kill idle
-  /// workers in FIFO order. The second element of std::pair is the time a worker becomes
-  /// idle.
-  std::list<std::pair<std::shared_ptr<WorkerInterface>, int64_t>> idle_of_all_languages_;
+  /// workers in FIFO order.
+  struct IdleWorkerEntry {
+    std::shared_ptr<WorkerInterface> worker;
+    // The time when the worker was last released by a task or freshly registered.
+    absl::Time idle_since;
+    // Don't kill this worker until this time. Clears on task assignment by setting to
+    // absl::InfinitePast();
+    absl::Time keep_alive_until;
```

Comment:
We can just have `keep_alive_util`?

Reply:
I know what you mean now. removing idle_since